api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -96,10 +98,19 @@ message DownstreamTlsContext {
 }
 
 // TLS context shared by both client and server TLS contexts.
-// [#next-free-field: 9]
+// [#next-free-field: 11]
 message CommonTlsContext {
   option (udpa.annotations.versioning).previous_message_type = "envoy.api.v2.auth.CommonTlsContext";
 
+  // Config for CA client plugin to get secrets
+  message CAClientPluginConfig {
```

Comment:
Please use `TypeExtensionConfig`, this is now in the v3 APIs.

Reply:
done

Reply:
you don't need `CAClientPluginConfig` message definition, `TypedExtensionConfig` contains name.

Reply:
The names are different: `CAClientPluginConfig.name` is the plugin name (used by the plugin architecture to select a plugin) and `TypedExtensionConfig.name` is the name of the extension (presumably used to provide type information?). No point overloading one name with 2 meanings.

Reply:
You shouldn't need a name to select the plugin. With `TypedExtensionConfig`, the type URL itself designates the plugin via the config type - plugin correspondence. The `name` in `TypedExtensionConfig` is opaque and arbitrary. This is a key feature of the v3 API.

Reply:
I didn't want to use the type_url to deduce plugin name. It will be good to support use cases where different plugins (implementations) share the same config-type. But the opaque `name` in `TypedExtensionConfig` can then be used as the plugin name. Is that the idea? I am okay with that if you don't see any issues.

Reply:
It's an xDS invariant that all extensions have a unique config type to support this. Do you want the CA client extension mechanism to be the one outlier? :)

Reply:
BTW, pls see my comment below about using this name similar to [name](https://github.com/envoyproxy/envoy/blob/master/api/envoy/extensions/transport_sockets/tls/v3/secret.proto#L33) in `SdsSecretConfig` i.e local cert vs validation-cert cases. If everyone agrees I will update the comment here to reflect what this name is for.

Reply:
Okay in that case I will eliminate the `CAClientPluginConfig` struct and use `TypedExtensionConfig` directly instead. So the config-type (type-url) is used to deduce the plugin to instantiate (I don't really like it but I will go along with it). In that case the `TypedExtensionConfig.name` can then be used to signal local cert vs CA cert similar to `SdsSecretConfig.name`. How is that?

Reply:
Sounds good. That's how typed extension configs are intended to be used in the API by design. Thanks.